src/core/image_utils.js

```diff
@@ -111,4 +113,99 @@ class NativeImageDecoder {
   }
 }
 
-export { NativeImageDecoder };
+const NUM_PAGES_THRESHOLD = 2;
+const MAX_IMAGES_TO_CACHE = 10;
+
+const GlobalImageCacheKind = {
+  NOT_CACHED: 1,
+  SHOULD_CACHE: 2,
+  IS_CACHED: 3,
+};
+
+class GlobalImageCache {
```

Comment:
Given that this class is important for correct image rendering, I would like to see unit tests being added for it. I think it's also the only way to get some test coverage for this patch and it should be relatively easy to test this class since we can get/set entries and assert that they are handled correctly.

Reply:
> I would like to see unit tests being added for it.

Well, I suppose that's a fair point; I'll see what I can come up with here :-)

> I think it's also the only way to get some test coverage for this patch

As mentioned in the commit message, since we're now handling repeated JPEG images differently there's *some* amount of test coverage already.
Although I suppose that complements, rather than replaces, dedicated unit-tests though...

Reply:
I've added some (high-level) unit-tests to ensure that the API receives the expected data, which I hope is useful as a complement to the reference tests. (I'm currently a bit short on time unfortunately.)

Reply:
I've added some (high-level) unit-tests to ensure that the API receives the expected image data, which I hope is useful as a complement to the reference tests. (I'm currently a bit short on time unfortunately.)